pretyping/evarsolve.ml

```diff
@@ -1344,7 +1362,7 @@ let solve_refl ?(can_drop=false) conv_algo env evd pbty evk argsv1 argsv2 =
   let args = Array.map2 (fun a1 a2 -> (a1, a2)) argsv1 argsv2 in
   let untypedfilter =
     restrict_upon_filter evd evk
-      (fun (a1,a2) -> conv_algo env evd Reduction.CONV a1 a2) args in
+      (fun (a1,a2) -> conv_algo false env evd Reduction.CONV a1 a2) args in
   let candidates = filter_candidates evd evk untypedfilter NoUpdate in
   let filter = closure_of_filter evd evk untypedfilter in
```

Comment:
To fix the FIXME about froze, it seem you just need to add a boolean "forbid pruning" and use it here

Reply:
Yes indeed, solve_refl succeeds only when there is no restriction needed.